src/sparseml/transformers/utils/sparse_model.py

```diff
@@ -93,6 +101,18 @@ def skip(*args, **kwargs):
                     zoo_stub=pretrained_model_name_or_path
                 )
 
+        # determine compression format, if any, from the model config
+        config = AutoConfig.from_pretrained(pretrained_model_name_or_path)
+        sparsity_config = getattr(config, SPARSITY_CONFIG_NAME, None)
+        if sparsity_config is not None:
+            format = sparsity_config.get("format")
+            sparsity_config = CompressionConfig.load_from_registry(
+                format, **sparsity_config
+            )
+            compressor = ModelCompressor.load_from_registry(
+                format, config=sparsity_config
+            )
+
         # temporarily set the log level to error, to ignore printing out long missing
         # and unexpected key error messages (these are EXPECTED for quantized models)
         logger = logging.getLogger("transformers.modeling_utils")
```

Comment:
Tried this out, but the runtime ended up being equivalent to what we currently have (probably because all of our keys are mismatched so no weights are loaded anyways) so I'm going to leave it out